server/android/src/android.rs

```diff
@@ -2,167 +2,89 @@
         let files_dir = PathBuf::from(&files_dir);
         let android_id: String = env.get_string(android_id).unwrap().into();
         let db_path = files_dir.join("omsupply-database");
-
         let cache_dir: String = env.get_string(cache_dir).unwrap().into();
 
-        let cert_path = files_dir.join("certs");
-        if !cert_path.join(PRIVATE_CERT_FILE).exists() {
-            generate_certs(&cert_path);
-        }
+        let settings = Settings {
+            server: ServerSettings {
+                port,
+                danger_allow_http: false,
+                debug_no_access_control: false,
+                cors_origins: vec!["http://localhost".to_string()],
+                base_dir: Some(files_dir.to_str().unwrap().to_string()),
+                machine_uid: Some(android_id),
+            },
+            database: DatabaseSettings {
+                username: "n/a".to_string(),
+                password: "n/a".to_string(),
+                port: 0,
+                host: "n/a".to_string(),
+                database_name: db_path.to_string_lossy().to_string(),
+                // See https://github.com/openmsupply/remote-server/issues/1076
+                init_sql: Some(format!("PRAGMA temp_store_directory = '{}';", cache_dir)),
+            },
+            // sync settings need to be configured at runtime
+            sync: None,
+            logging: None,
+        };
 
         // run server in background thread
         let thread = thread::spawn(move || {
-            actix_web::rt::System::new().block_on(async move {
-                let settings = Settings {
-                    server: ServerSettings {
-                        port,
-                        danger_allow_http: true,
-                        debug_no_access_control: false,
-                        cors_origins: vec!["http://localhost".to_string()],
-                        base_dir: Some(files_dir.to_str().unwrap().to_string()),
-                        machine_uid: Some(android_id),
-                    },
-                    database: DatabaseSettings {
-                        username: "n/a".to_string(),
-                        password: "n/a".to_string(),
-                        port: 0,
-                        host: "n/a".to_string(),
-                        database_name: db_path.to_string_lossy().to_string(),
-                        // See https://github.com/openmsupply/remote-server/issues/1076
-                        init_sql: Some(format!("PRAGMA temp_store_directory = '{}';", cache_dir)),
-                    },
-                    // sync settings need to be configured at runtime
-                    sync: None,
-                    logging: None,
-                };
-                let _ = start_server(settings, off_switch_receiver).await;
-            });
+            actix_web::rt::System::new()
+                .block_on(start_server(settings, off_switch_receiver))
+                .unwrap();
```

Comment:
Does this takes the app down when the start_server returns an error? Shouldn't happen but I would handle the error anyway.

Reply:
I haven't seen this happen while testing/developing (and I went through a lot of use case), I think it should be fine